ha/monitor/k8s/parser.py

```diff
@@ -156,13 +153,49 @@ def _create_health_alert(self, res_type, res_name, health_status, generation_id)
         self.event.set_specific_info({"generation_id": generation_id})
         return self.event.json
 
-    def parse(self, an_event, cached_state):
+    @staticmethod
+    def _get_resource_id(labels: dict, resource_id_map: dict) -> str:
+        """
+        get resource id
+
+        Check if pod name label exist if yes then check if resource map contains it
+        if resource id doesn't found then check if machine id label is exists if yes
+        then find resource id for same.
+
+        Note: machine id in label is different from the actual machine id inside pod
+        in file '/etc/machine-id' and the actual machine id is resource id.
+
+        Args:
+            labels (dict): pod labels with values
+            resource_id_map (dict): map of label (name/machine-id) with actual machine id
+            label 'cortx.io/pod_name' : uniq name (persist in reboot), key in resource map
+            label 'cortx.io/machine-id' : uniq id, key in resource map
+        Returns:
+            str: resource id which is nothing but actual machine id
+                inside pod in file '/etc/machine-id'
+        """
+        resource_id = None
+        # NOTE: All the cortx pods from the deployment that are being watched
+        # will have either label 'cortx.io/machine-id' or 'cortx.io/pod_name'
```

Comment:
pod_name label is actually "statefulset.kubernetes.io/pod-name"
(Ref: https://jts.seagate.com/browse/CORTX-30991?focusedCommentId=2367465&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2367465)

Reply:
done